src/Compilers/CSharp/Portable/FlowAnalysis/DefiniteAssignment.cs

```diff
@@ -2748,20 +2764,8 @@ public override void VisitForEachIterationVariables(BoundForEachStatement node)
                 int slot = GetOrCreateSlot(iterationVariable);
                 if (slot > 0) SetSlotAssigned(slot);
                 // NOTE: do not report unused iteration variables. They are always considered used.
-                NoteWrite(iterationVariable, null, read: true);
-            }
-        }
-
-        public override BoundNode VisitObjectInitializerMember(BoundObjectInitializerMember node)
-        {
-            var result = base.VisitObjectInitializerMember(node);
-
-            if ((object)_sourceAssembly != null && node.MemberSymbol != null && node.MemberSymbol.Kind == SymbolKind.Field)
-            {
-                _sourceAssembly.NoteFieldAccess((FieldSymbol)node.MemberSymbol.OriginalDefinition, read: false, write: true);
+                NoteWrite(iterationVariable, null, read: true, isRef: false);
```

Comment:
Does this behave correctly (e.g. is there any behavior change in the dataflow APIs) for a case like the following?

```cs
void M(Span<int> span) {
    foreach (ref var x in span) { }
}
```

Reply:
Similar response as above, there should be no change, since `iterationVariable` is a `LocalSymbol`, hence `isRef` has no effect on it.